pipemgr/src/pipemgr/models/concourse_handler.py

```diff
@@ -0,0 +1,66 @@
+from typing import Tuple
+
+from ..concourse.models import Session, LDAPTokenProvider, DiskTokenCache
+from ..concourse.utils import expect
+from .. import OUTPUT_ROOT
+
+
+CONCOURSE_URL = "https://concourse-v7.openstax.org"
+PIPELINE_URL = f"{CONCOURSE_URL}/api/v1/teams/CE/pipelines"
+PIPELINE_CONFIG_URL = f"{PIPELINE_URL}/{{pipeline}}/config"
+
+
+class ConcourseHandler:
```

Comment:
Why did you decide to use the singleton pattern here? I tend to stay away from singletons as they are often seen as code smells and are global.

Reply:
You are right: in this case Singletons are probably not needed since the program is pretty linear. Along with moving `ConcourseHandler` to the concourse package and renaming it `ConcourseService`, I will use a context manager and ditch the Singleton pattern.